onionmessage/endpoint.go

```diff
@@ -0,0 +1,256 @@
+package onionmessage
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"log/slog"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btclog/v2"
+	sphinx "github.com/lightningnetwork/lightning-onion"
+	"github.com/lightningnetwork/lnd/actor"
+	"github.com/lightningnetwork/lnd/fn/v2"
+	"github.com/lightningnetwork/lnd/lnutils"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/msgmux"
+	"github.com/lightningnetwork/lnd/record"
+	"github.com/lightningnetwork/lnd/subscribe"
+)
+
+// OnionMessageUpdate is onion message update dispatched to any potential
+// subscriber.
+type OnionMessageUpdate struct {
+	// Peer is the peer pubkey
+	Peer [33]byte
+
+	// PathKey is the route blinding ephemeral pubkey to be used for
+	// the onion message.
+	PathKey [33]byte
+
+	// OnionBlob is the raw serialized mix header used to relay messages in
+	// a privacy-preserving manner. This blob should be handled in the same
+	// manner as onions used to route HTLCs, with the exception that it uses
+	// blinded routes by default.
+	OnionBlob []byte
+
+	// CustomRecords contains any custom TLV records included in the
+	// payload.
+	CustomRecords record.CustomSet
+
+	// ReplyPath contains the reply path information for the onion message.
+	ReplyPath *lnwire.ReplyPath
+
+	// EncryptedRecipientData contains the encrypted recipient data for the
+	// onion message, created by the creator of the blinded route. This is
+	// the receiver for the last leg of the route, and the sender for the
+	// first leg uyp to the introduction point.
+	EncryptedRecipientData []byte
+}
+
+// OnionEndpointOption defines a function that can be used to configure
+// an OnionEndpoint. This allows for flexible configuration of the endpoint
+// when creating a new instance.
+type OnionEndpointOption func(*OnionEndpoint)
+
+// WithMessageServer sets the subscribe.Server for the OnionEndpoint.
+func WithMessageServer(server *subscribe.Server) OnionEndpointOption {
+	return func(o *OnionEndpoint) {
+		o.onionMessageServer = server
+	}
+}
+
+// OnionEndpoint handles incoming onion messages.
+type OnionEndpoint struct {
+	// subscribe.Server is used for subscriptions to onion messages.
+	onionMessageServer *subscribe.Server
+
+	// router is the sphinx router used to process onion_message_packet
+	router *sphinx.Router
+
+	// resolver is used to resolve node public keys from short channel IDs.
```

Comment:
this is needed because we can only send an onionmessag to a specific pubkey correct ?

Reply:
because I saw earlier that you allow channelID or Pubkey for the forwarding but I think we should not allow it or ?